packages/components/src/Actions/ActionDropdown/ActionDropdown.test.js

```diff
@@ -110,79 +111,67 @@ describe('InjectDropdownMenuItem', () => {
-		}
-		const divContainerElement = {
-			tagName: 'DIV',
-			classList: classListContainingClass,
-			getBoundingClientRect() {
-				return { top: 0, bottom: 35 };
-			},
-		};
-		const bodyElement = {
-			tagName: 'BODY',
-			classList: classListNotContainingClass,
-			getBoundingClientRect() {
-				return { top: 0, bottom: 35 };
-			},
-		};
-		function getActionDropdownEvent(top, bottom, container) {
-			return {
-				target: {
-					classList: classListContainingClass,
-					nextSibling: getDropdownMenuElement(top, bottom),
-					parentElement: {
-						classList: classListNotContainingClass,
-						// dropdown container
-						// simulate the tc-dropdown-container className
-						parentElement: container,
-					},
-				},
-			};
-		}
-		// dropdown that doesn't overflow
-		const event = getActionDropdownEvent(0, 25, divContainerElement);
-		// dropdown that is above the top of container
-		const topOverflowEvent = getActionDropdownEvent(-5, 10, divContainerElement);
-		// dropdown that is under the bottom of container
-		const bottomOverflowEvent = getActionDropdownEvent(20, 40, divContainerElement);
-		// dropdown that is under the bottom of body element
-		const bodyBottomOverflowEvent = getActionDropdownEvent(20, 40, bodyElement);
+		const isDropup = true;
+		const isDropdown = false;
 
-		const wrapper = shallow(<ActionDropdown items={[{ label: 'item 1' }, { label: 'item 2' }]} />);
-		expect(wrapper.state().dropup).toBeFalsy();
+		function testSwitch({ containerPosition, menuPosition, isInitialDropup, expectedDropupState }) {
+			// given
+			const container = document.createElement('div');
+			container.classList.add('tc-dropdown-container');
+			container.getBoundingClientRect = () => containerPosition;
 
-		// when / then
-		wrapper.props().onToggle(true, bottomOverflowEvent);
-		expect(wrapper.state().dropup).toBe(true);
+			const wrapper = mount(
+				<ActionDropdown
+					items={[{ label: 'item 1' }, { label: 'item 2' }]}
+					dropup={isInitialDropup}
+				/>,
+				{
+					attachTo: container,
+				},
+			);
+			container.querySelector('.dropdown-menu').getBoundingClientRect = () => menuPosition;
 
-		// when / then
-		wrapper.props().onToggle(true, topOverflowEvent);
-		expect(wrapper.state().dropup).toBeFalsy();
+			// when
+			wrapper
+				.find('button')
+				.first()
+				.simulate('click');
 
-		// when / then
-		wrapper.props().onToggle(true, event);
-		expect(wrapper.state().dropup).toBeFalsy();
+			// then
+			expect(container.querySelector('.dropdown').classList.contains('dropup')).toBe(
+				expectedDropupState,
+			);
+		}
 
-		// when / then
-		wrapper.props().onToggle(true, bodyBottomOverflowEvent);
-		expect(wrapper.state().dropup).toBe(true);
+		cases('dropdown/dropup switch', testSwitch, [
+			{
+				name: 'should dropup on dropdown bottom overflow',
+				containerPosition: { top: 0, bottom: 35 },
+				menuPosition: { top: 20, bottom: 40 },
+				isInitialDropup: false,
+				isDropupOrIsDropdown: isDropup,
```

Comment:
The props cases need to be updated too.
And the data corresponding can be directly given ; `isDropup` and `isDropdown`, even if I see now why they are, are not obvious, and misleading. Simple boolean without variables would avoid thinking :p